workflow/executor/common/common.go

```diff
@@ -80,6 +80,9 @@ func TerminatePodWithContainerID(c KubernetesClientInterface, containerID string
 		log.Infof("Container %s is already terminated: %v", container.ContainerID, container.State.Terminated.String())
 		return nil
 	}
+	if pod.Spec.ShareProcessNamespace != nil && *pod.Spec.ShareProcessNamespace {
+		return fmt.Errorf("cannot terminate a process-namespace-shared Pod %s", pod.Name)
```

Comment:
If we return immediately with an error during waiting sidecars here, the wait container exit with `0` and the pod succeeds. It's also the same in other error returns in this method too.

Reply:
test?

Reply:
Added a unit test of this method. I think we should have e2e tests for all the executors. If we had, [this functional test](https://github.com/argoproj/argo/blob/master/test/e2e/functional/sidecar-force-kill.yaml) found the problem.

Reply:
We have e2e smoke test for 3 of the executors?

Reply:
Ah, you're right. The problem occurs on the combination of k8sapi and `shareProcessNamespace`.